web/modules/custom/collection_page/collection_page.behat.inc

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * @file
+ * Contains \CollectionPageSubContext.
+ */
+
+use Behat\Gherkin\Node\TableNode;
+use Drupal\DrupalExtension\Context\DrupalSubContextBase;
+use Drupal\DrupalExtension\Context\DrupalSubContextInterface;
+
+/**
+ * Behat step definitions for testing collections.
+ */
+class CollectionPageSubContext extends DrupalSubContextBase implements DrupalSubContextInterface {
+  /**
+   * Test collections.
+   *
+   * @var \Drupal\node\Entity\Node[]
+   */
+  protected $collectionPages = [];
+
+  /**
+   * The bundle type of the content entity.
+   *
+   * @var String
+   */
+  protected $bundle = 'collection_page';
+
+  /**
+   * Creates content of a given type provided in the form.
+   * @codingStandardsIgnoreStart
+   * | title    | author     | status | created           |
+   * | My title | Joe Editor | 1      | 2014-10-17 8:00am |
+   * | ...      | ...        | ...    | ...               |
+   * @codingStandardsIgnoreEnd
+   *
+   * @Given content of type "Collection Page":
+   */
+  public function createNodes(TableNode $nodes) {
+    foreach ($nodes->getHash() as $hash) {
+      $node = (object) $hash;
+      $node->type = $this->bundle;
+      $this->nodeCreate($node);
+      $this->collectionPages[] = $node->nid;
+    }
+  }
```

Comment:
I know this is a duplicate but I implemented it this way only because the machine name of the entity bundle "Collection Page" is collection_page, so the "Given collection_page content:" would not be so user friendly.
I moved this to a generic behat file in the joinup profile so that we can store there functions that are useful and reusable.